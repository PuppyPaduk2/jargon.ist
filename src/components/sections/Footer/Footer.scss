@import '../../../styles/abstracts/index';

.c-footer {
  background-color: $secondary-color;
  color: $color-white;
  flex-shrink: 0;
  padding: pad(large) 0;

  &__inner {
    display: grid;
    grid-gap: px-to-rem(32px);
    grid-template-columns: 3fr 1fr 1fr;
  }

  &__logos {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
  }

  &__logo {
    &:hover {
      color: $primary-color;
      text-decoration: none;
    }
  }

  &__menu {
    list-style: none;
    margin: 0;
    padding: 0;
  }

  &__menu-item {
    margin-bottom: gap(2xsmall);
  }

  &__menu-link {
    color: $color-white;
  }
}
