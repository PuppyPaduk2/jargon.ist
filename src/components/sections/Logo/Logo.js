// @flow

import React from 'react';
import cx from 'classnames';

import logo from '../../../img/logo.svg';

import './Logo.scss';

type Props = {
  className?: string,
};

const Logo = ({ className }: Props) => (
  <span className={cx('c-logo', className)}>
    <img src={logo} alt="Jargon.ist" height="30" />
    <span className="c-logo__text">jargon.ist</span>
  </span>
);

Logo.defaultProps = {
  className: '',
};

export default Logo;
