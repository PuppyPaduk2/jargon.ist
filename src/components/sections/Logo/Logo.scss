@import '../../../styles/abstracts/index';

.c-logo {
  align-items: center;
  display: flex;
  flex-direction: row;

  &__text {
    font-size: px-to-rem(24px);
    line-height: px-to-rem(24px * $g-line-height-base);
    padding-left: pad(xsmall);
  }
}
