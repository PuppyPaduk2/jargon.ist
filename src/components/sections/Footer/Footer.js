// @flow

import React from 'react';
import Link from 'gatsby-link';

import { Container } from '../../ui';

import Logo from '../Logo';

import './Footer.scss';

const Footer = () => (
  <footer className="c-footer">
    <Container>
      <div className="c-footer__inner">
        <div>
          <Logo className="u-color-white" />
          <small>MIT lisansı ile geliştirilmiştir.</small>
        </div>

        <div>
          <h4 className="u-text-uppercase u-clear-gap-top u-gap-bottom-xsmall u-color-muted">
            Jargon
          </h4>
          <ul className="c-footer__menu">
            <li className="c-footer__menu-item">
              <Link to="/" className="c-footer__menu-link">
                Ana sayfa
              </Link>
            </li>
            <li className="c-footer__menu-item">
              <Link to="/dizin" className="c-footer__menu-link">
                Dizin
              </Link>
            </li>
            <li className="c-footer__menu-item">
              <Link to="/konular" className="c-footer__menu-link">
                Konular
              </Link>
            </li>
          </ul>
        </div>
        <div>
          <h4
            className="u-text-uppercase u-clear-gap-top u-gap-bottom-xsmall u-color-muted"
            lang="en"
          >
            Github
          </h4>
          <ul className="c-footer__menu">
            <li className="c-footer__menu-item">
              <a
                href={process.env.GATSBY_REPO_URL}
                className="c-footer__menu-link"
                target="_blank"
                rel="noopener noreferrer"
              >
                Kaynağı Görüntüle
              </a>
            </li>
            <li className="c-footer__menu-item">
              <a
                href={process.env.GATSBY_REPO_URL}
                className="c-footer__menu-link"
                target="_blank"
                rel="noopener noreferrer"
              >
                Katkıda Bulun
              </a>
            </li>
            <li className="c-footer__menu-item">
              <a
                href={`${process.env.GATSBY_REPO_URL || ''}/pulls?q=is:pr+is:open+label:new-jargon`}
                className="c-footer__menu-link"
                target="_blank"
                rel="noopener noreferrer"
              >
                Onay Bekleyen Jargonlar
              </a>
            </li>
            <li className="c-footer__menu-item">
              <a
                href={`${process.env.GATSBY_REPO_URL || ''}/issues/new`}
                className="c-footer__menu-link"
                target="_blank"
                rel="noopener noreferrer"
              >
                Sorun Bildir
              </a>
            </li>
            <li className="c-footer__menu-item">
              <a
                href={`${process.env.GATSBY_REPO_URL || ''}#katkida-bulunanlar`}
                className="c-footer__menu-link"
                target="_blank"
                rel="noopener noreferrer"
              >
                Katkıda Bulunanlar
              </a>
            </li>
          </ul>
        </div>
      </div>
    </Container>
  </footer>
);

export default Footer;
