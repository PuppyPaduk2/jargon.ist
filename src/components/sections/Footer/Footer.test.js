// @flow

import React from 'react';
import { shallow } from 'enzyme';

import Logo from '../Logo';

import Footer from './Footer';

it('renders correctly', () => {
  const wrapper = shallow(<Footer />);

  expect(wrapper).toMatchSnapshot();
});

it('renders logo', () => {
  const wrapper = shallow(<Footer />);

  expect(wrapper.find(Logo)).toHaveLength(1);
});

it('renders angolia logo', () => {
  const wrapper = shallow(<Footer />);

  expect(wrapper.find('img').prop('src')).toContain('algolia');
});
