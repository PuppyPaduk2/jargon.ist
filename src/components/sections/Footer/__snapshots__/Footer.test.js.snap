// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<footer
  className="c-footer"
>
  <Container
    className=""
  >
    <div
      className="c-footer__inner"
    >
      <div
        className="c-footer__logos"
      >
        <div>
          <Logo
            className="u-color-white"
          />
          <small>
            MIT lisansı ile geliştirilmiştir.
          </small>
        </div>
        <div>
          <img
            alt="Search by Algolia"
            height="18"
            src="https://www.algolia.com/static_assets/images/press/downloads/search-by-algolia-white.svg"
          />
        </div>
      </div>
      <div>
        <h4
          className="u-text-uppercase u-clear-gap-top u-gap-bottom-xsmall u-color-muted"
        >
          Jargon
        </h4>
        <ul
          className="c-footer__menu"
        >
          <li
            className="c-footer__menu-item"
          >
            <GatsbyLink
              className="c-footer__menu-link"
              to="/"
            >
              Ana sayfa
            </GatsbyLink>
          </li>
          <li
            className="c-footer__menu-item"
          >
            <GatsbyLink
              className="c-footer__menu-link"
              to="/dizin"
            >
              Dizin
            </GatsbyLink>
          </li>
          <li
            className="c-footer__menu-item"
          >
            <GatsbyLink
              className="c-footer__menu-link"
              to="/konular"
            >
              Konular
            </GatsbyLink>
          </li>
        </ul>
      </div>
      <div>
        <h4
          className="u-text-uppercase u-clear-gap-top u-gap-bottom-xsmall u-color-muted"
          lang="en"
        >
          Github
        </h4>
        <ul
          className="c-footer__menu"
        >
          <li
            className="c-footer__menu-item"
          >
            <a
              className="c-footer__menu-link"
              rel="noopener noreferrer"
              target="_blank"
            >
              Kaynağı Görüntüle
            </a>
          </li>
          <li
            className="c-footer__menu-item"
          >
            <a
              className="c-footer__menu-link"
              rel="noopener noreferrer"
              target="_blank"
            >
              Katkıda Bulun
            </a>
          </li>
          <li
            className="c-footer__menu-item"
          >
            <a
              className="c-footer__menu-link"
              href="/pulls?q=is:pr+is:open+label:new-jargon"
              rel="noopener noreferrer"
              target="_blank"
            >
              Onay Bekleyen Jargonlar
            </a>
          </li>
          <li
            className="c-footer__menu-item"
          >
            <a
              className="c-footer__menu-link"
              href="/issues/new"
              rel="noopener noreferrer"
              target="_blank"
            >
              Sorun Bildir
            </a>
          </li>
          <li
            className="c-footer__menu-item"
          >
            <a
              className="c-footer__menu-link"
              href="#katkida-bulunanlar"
              rel="noopener noreferrer"
              target="_blank"
            >
              Katkıda Bulunanlar
            </a>
          </li>
        </ul>
      </div>
    </div>
  </Container>
</footer>
`;
